src/stage1.rs

```diff
@@ -76,18 +75,65 @@ fn prepare_configs<P1: AsRef<Path>>(
     mig_info.update_config()?;
 
     // *********************************************************
-    // write network_manager filess to tmpfs
+    // write network_manager files to tmpfs
     let mut nwmgr_cfgs: u64 = 0;
     let nwmgr_path = path_append(work_dir, SYSTEM_CONNECTIONS_DIR);
+    let sys_proxy_copy_path = path_append(work_dir, SYSTEM_PROXY_DIR);
     create_dir_all(&nwmgr_path).upstream_with_context(&format!(
         "Failed to create directory '{}",
         nwmgr_path.display()
     ))?;
 
+    create_dir_all(&sys_proxy_copy_path).upstream_with_context(&format!(
+        "Failed to create directory '{}",
+        nwmgr_path.display()
+    ))?;
+
+    /* If migrating from balenaOS, copy all files from the system-connections file in /mnt/boot
+     * TODO: Check if we should copy them from the boot partition, or from the NM root overlay directory
+     */
+    if mig_info.os_name().starts_with(BALENA_OS_NAME) {
+        debug!("migrating from balenaOS - copying system-connections files");
+        let  nwmgr_files = read_dir(BALENA_SYSTEM_CONNECTIONS_BOOT_PATH).unwrap();
+        for path in nwmgr_files {
+            mig_info.add_nwmgr_file(path.unwrap().path());
+        }
+
+        debug!("migrating from balenaOS - copying system-proxy files");
+        let system_proxy_files = read_dir(BALENA_SYSTEM_PROXY_BOOT_PATH).unwrap();
+        for sys_proxy_path in system_proxy_files {
+            mig_info.add_system_proxy_file(sys_proxy_path.unwrap().path());
+        }
+    }
```

Comment:
It makes sense to copy all connections in a balenaOS-to-balenaOS migration... but we already some code related with migration of network connections. I don't know if that existing code can interfere with this new one in any way, but we should check this. In any case, I'd prefer to try to find a way to keep all this stuff together.

The code I am referring to lives in `wifi_configs.rs` and seems to be specific to Wifi connections, but still related.

Reply:
Hmm, we need to discuss this in the call. `wifi_config.rs`, `nwmgr_parser.rs` seem to be meant to for different types of functionality, like parsing NM/WPA Supplicant/Connman connection files, creating new ones, etc. We may want to disable that entirely for balenaOS to balenaOS migrations.

What I did in this part is to add the functionality of copying the existing system-proxy and system-connections to the tmpfs, in the case of migrating from balenaOS. This was and still being done for the connection files generated by `wifi_config.rs`, in the same place here in `stage1.rs.`

What I added doesn't interfere with the existing functionality, which still creates NM connection files apart from the ones I copy. The current way to disable creation of new NM connections by takeover is by using the existing '--no-nwmgr-check' flag.

Reply:
> What I added doesn't interfere with the existing functionality, which still creates NM connection files apart from the ones I copy. The current way to disable creation of new NM connections by takeover is by using the existing '--no-nwmgr-check' flag.

Yes, I get this @acostach ! To be clear, my concern is not about this being incorrect, it's all about trying to make the code as whole more consistent to make longer-term easier. And also: I don't think the existing code was particularly to follow (maybe it's just me?), so I don't expect us to do any miracles here -- but I am still hopeful we can make it a bit more consistent.

I created this [commit sketching of a suggestion](https://github.com/balena-os/takeover/commit/bbe6fd67652f9f3e1ed7f459907f7b9caf24eabf). This one is not meant to be cherry-picked! It's not tested. I created the commit just because I think it's easier than trying to explain in words. But the idea was to try to make the new code more aligned with the code we already had for handling WiFi networks:

1. In `migrate_info.rs` populate the `MigrateInfo` fields. [Like we did before](https://github.com/balena-os/takeover/blob/f2ed41a7703db195e5427dcfe1a184e5820b5295/src/stage1/migrate_info.rs#L127) for `wifis`.
2. In `stage1.rs` do the actual copying. Again, this is similar to what we did for `wifis`: [this is the same place](https://github.com/balena-os/takeover/blob/f2ed41a7703db195e5427dcfe1a184e5820b5295/src/stage1.rs#L103
) we currently already create the connection files for WiFi.

(As a bonus suggestion, I also tossed in some smaller changes to propagate the errors up the call stack instead of using `unwrap()`, which panics in case of errors. Just for more friendly error reporting.)

In summary, what I am proposing is not a big change, but I think it helps to keep similar operations closer together in the code, which hopefully will help future maintainers to understand it.

Reply:
@acostach , I was just now reviewing our Project on Fibery. Sorry, I didn't recall we had a separate task to deal with this "network migration" point.

From my perspective, it would be OK to merge this PR as it is right now and tackle this remaining point on a different PR. I guess we'll talk about this on the upcoming sync, but leaving a comment for posterity!

Reply:
So, as agreed on our sync, I am marking this point as resolved and approving the PR! :rocket: